godot-core/src/obj/call_deferred.rs

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use crate::builtin::{Callable, Variant};
+use crate::meta::UniformObjectDeref;
+use crate::obj::bounds::Declarer;
+use crate::obj::GodotClass;
+#[cfg(since_api = "4.2")]
+use crate::registry::signal::ToSignalObj;
+use godot_ffi::is_main_thread;
+use std::ops::DerefMut;
+
+// Dummy traits to still allow bounds and imports.
+#[cfg(before_api = "4.2")]
+pub trait WithDeferredCall<T: GodotClass> {}
+
+/// Trait that is automatically implemented for engine classes and  user classes containing a `Base<T>` field.
+///
+/// This trait enables type safe deferred method calls.
+///
+/// # Usage
+///
+/// ```no_compile
+/// # use godot::prelude::*;
+/// # use godot::classes::CollisionShape2D;
+/// # use std::f32::consts::PI;
+/// fn some_fn(mut shape: Gd<CollisionShape2D>)
+/// {
+///     shape.apply_deferred(|shape_mut| shape_mut.rotate(PI))
+/// }
+/// ```
+#[cfg(since_api = "4.2")]
```

Comment:
The whole file is pointless before Godot 4.2, so this should rather be an inner attribute `#![cfg(...)]` at the start of the file (after license header).

Reply:
I see, in that case it makes sense.

Either way, we're likely going to remove Godot 4.1 support in godot-rust v0.4, so this soon won't matter. So don't spend too much time on it :slightly_smiling_face: